django/utils/datastructures.py

```diff
@@ -170,8 +182,21 @@ def setlistdefault(self, key, default_list=None):
             # another value -- QueryDict.setlist() does. Look it up.
         return self._getlist(key)
 
+    def pop(self, key, *args):
+        self._assert_mutable()
+        return super().pop(key, *args)
+
+    def popitem(self):
+        self._assert_mutable()
+        return super().popitem()
+
+    def clear(self):
+        self._assert_mutable()
+        super().clear()
```

Comment:
As `QueryDict` inherits `MultiValueDict`, you might be able to remove/refactor some of the code there. 
Might be more than just these